src/types.d.ts

```diff
@@ -43,6 +43,35 @@ export interface EnterpriseLearnerData {
   staffEnterpriseCustomer: Types.EnterpriseCustomer;
 }
 
+interface DueDate {
+  name?: string;
+  date?: string;
+  url?: string;
+}
+
+export interface EnterpriseCourseEnrollment {
+  course_run_id: string;
+  course_key: string;
+  course_type: string;
+  org_name: string;
+  course_run_status: string;
+  display_name: string;
+  emails_enabled?: boolean;
+  certificate_download_url?: string;
+  created: string;
+  start_date?: string;
+  end_date?: string;
+  mode: string;
+  is_enrollment_active: boolean;
+  product_source: string;
+  enroll_by?: string;
+  pacing: boolean;
+  course_run_url: string;
+  resume_course_run_url?: string;
+  is_revoked: boolean;
+  due_dates: (DueDate | {})[]
```

Comment:
Since due dates allow empty values for the object children, it covers the the schema for when a due date exists, and if it allows empty
due dates src: https://github.com/openedx/edx-platform/blob/9cade7a3488eee24109311cb145747a01348df14/lms/djangoapps/course_api/api.py#L245
serializer src: https://github.com/openedx/enterprise-access/blob/106b52bb8632ef691ae3b884e280ab2a4175eef6/enterprise_access/apps/bffs/serializers.py#L279C4-L282C6